pandas/core/sparse/frame.py

```diff
@@ -191,11 +191,13 @@ def _init_spmatrix(self, data, index, columns, dtype=None,
         for col, rowvals in values.groupby(data.col):
             # get_blocks expects int32 row indices in sorted order
             rows = rowvals.index.values.astype(np.int32)
+            vals = np.array([y for x, y in sorted(rowvals.iteritems())],
```

Comment:
Which index do you mean? This code should sort the values by the index of rowvals, or am I mistaken?
Sorry, I'm new to the codebase, so I might be misunderstanding something.